transport/src/main/java/io/netty5/channel/nio/AbstractNioChannel.java

```diff
@@ -48,6 +51,75 @@
     boolean readPending;
     private final Runnable clearReadPendingRunnable = this::clearReadPending0;
 
+    private final NioProcessor nioProcessor = new NioProcessor() {
+        @Override
+        public void register(Selector selector) throws ClosedChannelException {
+            int interestOps;
+            SelectionKey key = selectionKey;
+            if (key != null) {
+                interestOps = key.interestOps();
+                key.cancel();
+            } else {
+                interestOps = 0;
+            }
+            selectionKey = javaChannel().register(selector, interestOps, this);
```

Comment:
This will make it harder to forward port #12610

Reply:
Yep ... that said I think we should do this differently in 5.0 anyway . So let's worry about that later